src/core/classes/contract-wrapper.ts

```diff
@@ -100,23 +99,39 @@ export class ContractWrapper<
    * @internal
    */
   public async getCallOverrides(): Promise<CallOverrides> {
-    const chainId = await this.getChainID();
-    const speed = this.options.gasSettings?.speed || "fastest";
-    const maxGasPrice = this.options.gasSettings?.maxPriceInGwei || 300;
-    const gasPriceChain = await getGasPriceForChain(
-      chainId,
-      speed,
-      maxGasPrice,
-    );
-    if (!gasPriceChain) {
-      return {};
-    }
-    // TODO: support EIP-1559 by try-catch, provider.getFeeData();
     return {
-      gasPrice: ethers.utils.parseUnits(gasPriceChain.toString(), "gwei"),
+      gasPrice: await this.getPreferredGasPrice(),
```

Comment:
by setting `gasPrice` you'd always override the transaction type to be 1 instead of 2.

for setting transaction type 2, you'd need to do https://github.com/thirdweb-dev/contracts/blob/v2/scripts/deploy/testDeployV2Setup.ts#L43-L47

our sdk needs to work with both because fantom/avax are only supporting pre-eip1559.  polygon/eth are post-eip1559